extensions/wikia/PortableInfobox/tests/XmlParserTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+require_once( dirname( __FILE__ ) . '/../services/Parser/XmlParser.php' );
+
+class TestParser implements \Wikia\PortableInfobox\Parser\ExternalParser {
+	public function parse( $text ) {
+		return "parse($text)";
+	}
+
+	public function parseRecursive( $text ) {
+		return "parseRecursive($text)";
+	}
+}
+
+class PortableInfoboxParserNodesTest extends WikiaBaseTest {
+
+	protected function setUp() {
+		parent::setUp();
+		require_once( dirname( __FILE__ ) . '/../PortableInfobox.setup.php' );
+		foreach ( $wgAutoloadClasses as $class => $file ) {
+			require_once( $file );
+		}
+	}
+
+	public function testIsEmpty() {
+		$parser = new \Wikia\PortableInfobox\Parser\XmlParser([
+			'elem2' => 'ELEM2',
+			'lado2' => 'LALALA'
+		]);
+		$markup = '
+			<infobox>
+				<comparison>
+				   <set>
+					  <header><value>Combatientes</value></header>
+					  <pair source="lado1" />
+					  <pair source="lado2" />
+				   </set>
+				</comparison>
+			</infobox>
+		';
+		$data = $parser->getDataFromXmlString( $markup );
+		// infobox -> comparison -> set -> header
+		$this->assertTrue( $data[0]['data']['value'][0]['value'][0]['isEmpty'] == false );
```

Comment:
The messages from comments could be passed as message to the assertions, giving immediate information about reason in case of failure